src/modules/Game.class.js

```diff
@@ -1,68 +1,265 @@
+
+        for (let c = 0; c < 4; c++) {
+          if (this.state[r][c] !== 0) {
+            values.push(this.state[r][c]);
+          }
+        }
+
+        for (let i = 0; i < values.length; i++) {
+          if (values[i] === values[i + 1]) {
+            values[i] *= 2;
+            values[i + 1] = 0;
+            this.score += values[i];
+            isMovable = true;
+          }
+        }
+
+        const updatedRow = values.filter((value) => value !== 0);
+
+        while (updatedRow.length < 4) {
+          updatedRow.push(0);
+        }
+
+        for (let c = 0; c < 4; c++) {
+          if (this.state[r][c] !== updatedRow[c]) {
+            this.state[r][c] = updatedRow[c];
+            isMovable = true;
+          }
+        }
+      }
+
+      if (isMovable) {
+        this.getRandomCells();
+        this.checkGameStatus();
+      }
+    }
+  }
+  moveRight() {
+    if (this.status === Game.Status.playing) {
+      let isMovable = false;
+
+      for (let r = 0; r < 4; r++) {
+        const values = [];
+
+        for (let c = 3; c >= 0; c--) {
+          if (this.state[r][c] !== 0) {
+            values.push(this.state[r][c]);
+          }
+        }
+
+        for (let i = 0; i < values.length; i++) {
+          if (values[i] === values[i + 1]) {
+            values[i] *= 2;
+            values[i + 1] = 0;
+            this.score += values[i];
+            isMovable = true;
+          }
+        }
+
+        const updatedRow = values.filter((value) => value !== 0);
+
+        while (updatedRow.length < 4) {
+          updatedRow.push(0);
+        }
+
+        for (let c = 0; c < 4; c++) {
+          if (this.state[r][c] !== updatedRow[3 - c]) {
+            this.state[r][c] = updatedRow[3 - c];
+            isMovable = true;
+          }
+        }
+      }
+
+      if (isMovable) {
+        this.getRandomCells();
+        this.checkGameStatus();
+      }
+    }
+  }
+  moveUp() {
+    if (this.status === Game.Status.playing) {
+      let isMovable = false;
+
+      for (let c = 0; c < 4; c++) {
+        const values = [];
+
+        for (let r = 0; r < 4; r++) {
+          if (this.state[r][c] !== 0) {
+            values.push(this.state[r][c]);
+          }
+        }
+
+        for (let i = 0; i < values.length; i++) {
+          if (values[i] === values[i + 1]) {
+            values[i] *= 2;
+            values[i + 1] = 0;
+            this.score += values[i];
+            isMovable = true;
+          }
+        }
+
+        const updatedColumn = values.filter((value) => value !== 0);
+
+        while (updatedColumn.length < 4) {
+          updatedColumn.push(0);
+        }
+
+        for (let r = 0; r < 4; r++) {
+          if (this.state[r][c] !== updatedColumn[r]) {
+            this.state[r][c] = updatedColumn[r];
+            isMovable = true;
+          }
+        }
+      }
+
+      if (isMovable) {
+        this.getRandomCells();
+        this.checkGameStatus();
+      }
+    }
   }
+  moveDown() {
```

Comment:
Arthur, the AI is right, all move methods look way too similar to be separated code.
Can you manage to create a single function that accepts an enum ("right", "left", "up", "down") and manages the minimal changes accordingly to it?